src/screens/lock/view/LockScreenBars.tsx

```diff
@@ -28,12 +28,12 @@ export const FlagshipBars = ({
   } = UIState
 
   useEffect(() => {
-    flagshipUI.on('change', (state: NormalisedFlagshipUI) => {
+    lockScreenUi.on('lockscreen:change', (state: NormalisedFlagshipUI) => {
       setUIState({ ...UIState, ...state })
```

Comment:
By using `setUIState(oldState => ({...oldState, ...state}))` you can remove the hook dependency to UIState and so reduce the amount of call to `.removeAllListeners()`/`.on()`

Reply:
not sure here, where do we get the `oldState`?

Reply:
this is provided by the useState API: https://reactjs.org/docs/hooks-reference.html#usestate (search for `prevState`)

`setState` can get a value, or a function callback that takes the previous state in argument

Reply:
yea good catch